app/lib/features/pins/pages/pin_details_page.dart

```diff
@@ -0,0 +1,286 @@
+    );
+  }
+
+  Widget _buildBodyUI() {
+    final pinData = ref.watch(pinProvider(widget.pinId));
+    return pinData.when(
+      data: (pin) {
+        manageBackwardPinLinkSupport(context, ref, pin);
+        return SingleChildScrollView(
+          child: Column(
+            children: [
+              const SizedBox(height: 20),
+              _buildPinHeaderUI(pin),
+              const SizedBox(height: 20),
+              AttachmentSectionWidget(manager: pin.attachments()),
+              const SizedBox(height: 20),
+              CommentsSection(manager: pin.comments()),
+            ],
+          ),
+        );
+      },
+      loading: () => Skeletonizer(child: Text(L10n.of(context).loadingPin)),
+      error: (err, st) => Text(
+        L10n.of(context).errorLoadingPin(err),
+      ),
+    );
+  }
+
+  // pin actions menu builder
+  Widget _buildActionMenu() {
+    final pinData = ref.watch(pinProvider(widget.pinId));
+    return pinData.when(
+      data: (pin) {
+        return PopupMenuButton<String>(
+          key: PinDetailsPage.actionMenuKey,
+          icon: const Icon(Atlas.dots_vertical_thin),
+          itemBuilder: (context) => _buildAppBarActionMenuItems(pin),
+        );
+      },
+      loading: () => Skeletonizer(child: Text(L10n.of(context).loadingPin)),
+      error: (err, st) => const SizedBox.shrink(),
+    );
+  }
+
+  List<PopupMenuEntry<String>> _buildAppBarActionMenuItems(ActerPin pin) {
+    List<PopupMenuEntry<String>> actions = [];
+
+    //Get my membership details
+    final membership =
+        ref.watch(roomMembershipProvider(pin.roomIdStr())).valueOrNull;
+    if (membership != null) {
+      //Check for can post pin permission
+      if (membership.canString('CanPostPin')) {
+        //EDIT PIN TITLE MENU ITEM
+        actions.add(
+          PopupMenuItem<String>(
+            key: PinDetailsPage.editBtnKey,
+            onTap: () => showEditPintTitleDialog(context, ref, pin),
+            child: Text(L10n.of(context).editTitle),
+          ),
+        );
+
+        //EDIT PIN DESCRIPTION MENU ITEM
+        actions.add(
+          PopupMenuItem<String>(
+            key: PinDetailsPage.editBtnKey,
+            onTap: () => showEditPintDescriptionDialog(context, ref, pin),
+            child: Text(L10n.of(context).editDescription),
+          ),
+        );
+      }
+
+      final canRedact = ref.watch(canRedactProvider(pin));
+      if (canRedact.valueOrNull == true) {
+        final roomId = pin.roomIdStr();
+        //DELETE PIN MENU ITEM
+        actions.add(
+          PopupMenuItem<String>(
+            onTap: () => showRedactDialog(
+              context: context,
+              pin: pin,
+              roomId: roomId,
+            ),
+            child: Text(
+              L10n.of(context).removePin,
+              style: TextStyle(color: Theme.of(context).colorScheme.error),
+            ),
+          ),
+        );
+      } else {
+        //REPORT PIN MENU ITEM
```

Comment:
not an else, even if you _can_ redact, you can still report. always.
```suggestion
        //REPORT PIN MENU ITEM
```

and in that case report should come before redact.